services:
  grafana:
    image: grafana/grafana:10.3.1
    restart: unless-stopped
    ports: 
      - "9000:3000"
    environment:
      GF_SERVER_ROOT_URL: "%(protocol)s://%(domain)s:%(http_port)s/"
      GF_AUTH_ANONYMOUS_ENABLED: "true"
    volumes:
      - ./grafana/prometheus-datasource.yaml:/etc/grafana/provisioning/datasources/prometheus-datasource.yaml
      - ./grafana/dashboards-config.yaml:/etc/grafana/provisioning/dashboards/default.yaml
      - ./grafana/dashboards:/var/lib/grafana/dashboards
    depends_on:
      - prometheus
    networks:
      - monitoring-net
  prometheus:
    image: prom/prometheus:v2.45.3
    restart: unless-stopped
    labels:
      - docker-volume-backup.stop-during-backup=true
    ports:
      - "9090:9090"
    volumes:
      - ./prometheus.yaml:/etc/prometheus/prometheus.yml:ro
      - prometheus-data:/prometheus
    depends_on:
      - cadvisor
    extra_hosts:
      - "host.docker.internal:host-gateway"
    networks:
      - monitoring-net
  cadvisor:
    image: gcr.io/cadvisor/cadvisor:v0.47.2
    restart: unless-stopped
    ports:
    - 9091:8080
    volumes:
    - /:/rootfs:ro
    - /var/run:/var/run:rw
    - /sys:/sys:ro
    - /var/lib/docker/:/var/lib/docker:ro
    networks:
      - monitoring-net
volumes:
  prometheus-data:
    external: true
networks:
  monitoring-net:
    external: true
